alicloud/resource_alicloud_cdn_domain.go

```diff
@@ -0,0 +1,751 @@
+						},
+						"custom_page_url": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"error_code": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+
+			"refer_config": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"refer_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "block",
+							ValidateFunc: validateCdnReferType,
+						},
+						"refer_list": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"allow_empty": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "on",
+							ValidateFunc: validateCdnEnable,
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+
+			"auth_config": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"auth_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "no_auth",
+							ValidateFunc: validateCdnAuthType,
+						},
+						"master_key": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCdnAuthKey,
+						},
+						"slave_key": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validateCdnAuthKey,
+						},
+						"time_out": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "1800",
+						},
+					},
+				},
+				MaxItems: 1,
+			},
+
+			"http_header_config": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"header_key": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateCdnHttpHeader,
+						},
+						"header_value": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"header_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				MaxItems: 10,
+			},
+
+			"cache_config": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cache_content": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ttl": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"cache_type": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateCacheType,
+						},
+						"weight": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "1",
```

Comment:
modify weight's type to int and set its using ValidateFunc as validateIntegerInRange